tools/ci_build/build.py

```diff
@@ -1325,10 +1327,13 @@ def generate_build_tree(
     if args.use_snpe:
         cmake_args += ["-Donnxruntime_USE_SNPE=ON"]
 
+    # minor note: the below condition basically just means an apple framework build
+    # and it can cover platforms including macosx/iphoneos/iphonesimulator/maccatalyst.
     if args.build_apple_framework or args.ios:
-        if not args.cmake_generator == "Xcode":
+        # Note: Xcode CMake generator doesn't have a good support for Mac Catalyst yet.
+        if not args.mac_catalyst and not args.cmake_generator == "Xcode":
             raise BuildError(
-                "iOS/MacOS framework build requires use of the Xcode CMake generator ('--cmake_generator Xcode')."
+                "iOS/MacOS framework build requires use of the Xcode CMake generator ('--cmake_generator Xcode'). (catalyst builds not included.)"
```

Comment:
From the comment it sounds like the requirement is: 
- if mac_catalyst, cmake_generator must not be Xcode
- if not mac_catalyst, cmake_generator must be Xcode

But nothing is checking the first condition. 

nit: not sure the `(catalyst builds not included.)` is necessary as it could add confusion given the user isn't trying to do a mac_catalyst build. 

Would be good to document the cmake_generator requirement in the help for `--ios` and `--mac_catalyst` etc., and maybe set cmake_generator (if not specified) by default to a valid value for the build.